### PR TITLE
Implements move_to

### PR DESCRIPTION
**Pull request by hbcarlos**
Hi @Waidhoferj.

I noticed the move feature had not yet been implemented here in the python bindings. I trying to implement it since we need it for JupyterLab.

Can you take a look at this? It is the first time I have looked at y-rs and y-py, and the first time I used rust, so I'm unsure if I'm implementing it correctly.

I'm keeping the PR as a draft because I want to implement `move_range_to` in the same PR, but first, I would like you to take a look at the `move_to` function in case I'm doing something wrong.

### CODE REVIEW

File: src/y_array.rs
Comment (Waidhoferj) [inline src/y_array.rs:186]:
```suggestion
Err(PyIndexError::default_message())
```
Including a default message or specifically referencing whether `source` or `target` is out of bounds would be helpful.

**Review by Waidhoferj — COMMENTED** [review]
Hi @hbcarlos, thanks for adding move functionality! This looks good so far. Let me know if you have any questions about the codebase.

@Horusiath could you double check this?

**Comment by hbcarlos** [comment]
Hey @Waidhoferj! Thank you so much for the help!

I added a new commit with your suggestion and the implementation of `move_range_to`.

**Comment by Waidhoferj** [comment]
Hi @hbcarlos, looks good! Could you add test cases covering these functions to `test_y_array.py`?

**Comment by hbcarlos** [comment]
I added some tests in my last commit.

**Comment by SylvainCorlay** [comment]
Ping @Waidhoferj. Should we be able to move forward with this?

**Review by Waidhoferj — APPROVED** [review]
Tests look great! Feel free to merge.

**Comment by hbcarlos** [comment]
Thanks, @Waidhoferj! I don't have maintainer rights, I can not merge it.

**Pull request merged**
